contracts/Gate.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+pragma solidity 0.7.1;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+
+import "./interfaces/IGate.sol";
+import "./interfaces/IERC1155.sol";
+
+/**
+ * @title Gate contract between Boson router and ERC1155NonTransferable
+ * Enables conditional commit
+ */
+
+contract Gate is IGate, Ownable, Pausable {
+    mapping(uint256 => uint256) private voucherToToken;
+    mapping(address => mapping(uint256 => bool)) private isDeactivated; // mapping user => voucherSet => bool
+
+    IERC1155 private nonTransferableTokenContract;
+    address private bosonRouterAddress;
+  
+    modifier onlyFromRouter() {
+        require(bosonRouterAddress != address(0), "UNSPECIFIED_BR"); //hex"20" FISSION.code(FISSION.Category.Find, FISSION.Status.NotFound_Unequal_OutOfRange)
```

Comment:
Instead of doing this here, we could be setting it in a constructor and ensuring that it is not `address(0)` and checking for it not being address(0) in *setBosonRouterAddress* .... saving on gas on every call made to a function with the modifier onlyFromRouter(). Perhaps this is an enhancement.

Reply:
That's true. I followed the existing implementation from `VoucherKernel`. Probably it should be updated there too, since that is call even more frenquetly than gate contract.
I will change this in next commit.